### PR TITLE
TLM: Create unit tests for status/state converters

### PR DESCRIPTION
**Pull request by robertasn**
## Description
Unit tests for status/state converters have been created

### CODE REVIEW

File: test/src/telemetry/writerConvertStateMachineState.test.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+struct WriterConvertStateMachineState : public ::testing::Test {
+ protected:
+  void SetUp() {}
+  void TearDown() {}
+};
+
+TEST_F(WriterConvertStateMachineState, idleState)
+{
+  ASSERT_EQ(Writer::convertStateMachineState(State::kIdle), "IDLE");
```
Comment (DonaldJennings) [inline test/src/telemetry/writerConvertStateMachineState.test.cpp]: You should add some error messages at the end of each ASSERT() for the case that the test fails then we can get a meaningful message with it. [Refer to this with Error messages](https://hyp-ed.github.io/styleguide/#Error%20Messages)

File: test/src/telemetry/writerConvertStateMachineState.test.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+#include "gtest/gtest.h"
+#include "telemetry/writer.hpp"
+#include "data/data.hpp"
+
+using namespace hyped::telemetry;
+using namespace hyped::data;
+
+struct WriterConvertStateMachineState : public ::testing::Test {
```
Comment (DonaldJennings) [inline test/src/telemetry/writerConvertStateMachineState.test.cpp]: This should have a comment explaining what these tests are. Comments may not be needed for the actual tests but the struct should always have one : )

File: test/src/telemetry/writerConvertStateMachineState.test.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Author: Robertas Norkus
+ * Organisation: HYPED
+ * Date:
+ * Description:
+ *
+ *    Copyright 2019 HYPED
```
Comment (DonaldJennings) [inline test/src/telemetry/writerConvertStateMachineState.test.cpp]: This should probably be 2021 HYPED.

File: test/src/telemetry/writerConvertStateMachineState.test.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  void SetUp() {}
+  void TearDown() {}
+};
+
+TEST_F(WriterConvertStateMachineState, idleState)
+{
+  ASSERT_EQ(Writer::convertStateMachineState(State::kIdle), "IDLE");
+}
+
+TEST_F(WriterConvertStateMachineState, calibratingState)
```
Comment (DonaldJennings) [inline test/src/telemetry/writerConvertStateMachineState.test.cpp]: Try make the test names more meaningful . Test names should always begin with a verb such as "handles" or "accepts" to name a few examples. A good one for this case could be "converts" so the name of this particular test could be "convertsCalibratingState"

File: test/src/telemetry/writerConvertModuleStatus.test.cpp [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Author: Robertas Norkus
+ * Organisation: HYPED
+ * Date:
+ * Description:
+ *
+ *    Copyright 2019 HYPED
```
Comment (DonaldJennings) [inline test/src/telemetry/writerConvertModuleStatus.test.cpp]: Same as other comment. This should be 2021 HYPED and the Date and description should be filled out

File: test/src/telemetry/writerConvertStateMachineState.test.cpp (hunk 3 above)
Comment (DonaldJennings) [reply test/src/telemetry/writerConvertStateMachineState.test.cpp]: Date and Description should also be filled out

**Review by DonaldJennings — CHANGES_REQUESTED** [review]
Just a few minor changes to make which centre around the Testing style guide. Make sure all asserts have an appropriate error message in case of failing and that the test names begin with a verb.

**Comment by mifrandir** [comment]
The file names seem a bit off to me. I think we are using snake_case everywhere else.

**Comment by DonaldJennings** [comment]
@MiltFra yea just noticed that as well the general naming for files should be snake_case

**Review by DonaldJennings — APPROVED** [review]

**Review by pmorande27 — APPROVED** [review]
Seems okay as a starting point for testing :)

**Pull request merged**